### PR TITLE
test: Refactor Cumulocity IoT device profile system tests

### PR DESCRIPTION
**Pull request by reubenmiller**
## Proposed changes

Refactor the Cumulocity IoT device profile system tests to use keywords provided in the newer version of [robotframework-c8y](https://github.com/reubenmiller/robotframework-c8y).

Note: The creation of the device profile and applying it to a device has been separated into two keywords to enable improved control, and the cleanup of the device profile (e.g. deletion from the repository) happens automatically in the `Create Device Profile` keyword.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: tests/RobotFramework/tests/cumulocity/software_management/sm-plugin.robot [hunk 1]
```diff
@@ -76,6 +76,7 @@ sm-plugins download files from Cumulocity
     Should Be Equal    ${downloaded}    Testing a thing
 
 Filter packages list using include pattern
+    Execute Command    sudo tedge config unset software.plugin.exclude
```
Comment (rina23q) [inline tests/RobotFramework/tests/cumulocity/software_management/sm-plugin.robot:79]: Why is this line added? It seems the setup is done in test level.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Review by rina23q — APPROVED** [review]
One minor question. Others look good

**Review by albinsuresh — APPROVED** [review]
LGTM. Glad that you split the creation and application of the device profile with implicit cleanup. Explicit teardown was very error-prone with the earlier logic.
